routemaster/services/deliver.rb

```diff
@@ -36,14 +45,24 @@ def call
         error = nil
         start_at = Routemaster.now
         begin
-        # send data
+          _maybe_throttle!
+          @subscriber.attempting_delivery(start_at)
+
+          # send data
           response = _conn.post do |post|
             post.headers['Content-Type'] = 'application/json'
             post.body = Oj.dump(_data, mode: :compat)
           end
-          error = CantDeliver.new("HTTP #{response.status}") unless response.success?
+
+          unless response.success?
+            @subscriber.change_health_by(-2)
+            error = CantDeliver.new("HTTP #{response.status}")
+          end
+        rescue EarlyThrottle => e
+          error = _wrap_error(e)
         rescue Faraday::Error::ClientError => e
-          error = CantDeliver.new("#{e.class.name}: #{e.message}")
+          @subscriber.change_health_by(-2)
```

Comment:
Well, the `-2` is applied here because the `+1` is applied here too. The Throttle should not try to modify the HP of a subscriber.